### ISSUE TITLE
Add a "simple planning mode" to replace DSU

### ISSUE THREAD

**Issue by gfgtdf** [wesnoth/wesnoth#7507]
### Describe the desired feature

DSU has a few drawbacks:
1) Its complicates the mp sync mechanics quite a lot, since the commit vision command reevaluated the undo stack, the game now has to make sure the undo stack is the same on all clients.
2) We have to make sure dsu is disabled when making a ai handle a side since the ai cannot handle it.
3) It might lead break scenarios when a sighted even is now fired much later that expected.
4) While it guards the player from missclicks in some cases, its still very much possible that move cannot be undone.

Planning mode, which is supposed to replace DSU, is considered to be too complicated by some players, so the proposal is to add a "plannign mode light", it should:
1) Not draw many arrows
2) Not allow planning of multiple-turn moves.
3) Still show a clear indicator, that the current gamestate is only the "planned gamestate" is that's the case.
4) Probably autoamticially end itself when attacking a unit or recruiting a unit.

**Comment by knyghtmare** [comment]
What is DSU? 😨

**Comment by gfgtdf** [comment]
"Delayed Shroud Updates"

**Comment by gfgtdf** [comment]
> I haven't looked at the DSU code, but would it simplify the code while still being a reasonable UI to force an immediate shroud redraw if a WML or Lua sighted event should be triggered?

I don't think this would make it simpler, Other events might still depend on which tile is visible, (filter_vision) and we still have the same problem. Im currently thinking making a special exception for sighted event would just make it it even more complicated.

Furthermore, this would allow players to "fish" for sighted events  (not a big problem, but should be noted)

> shroud redraw

I think you know but just to make it sure: This is not about the drawing code, DSU is about on purpose not uncovering tiles to allow undoing of moves.
